### PR TITLE
Delays automagdrops by half a second.

### PR DESCRIPTION
**Pull request by ultraplen**
#13455 but with the delay being done in-code. Now also delays the actual magdrop for the same amount of time as the sound to increase immulsions.

:cl: 

 * tweak: Delayed the automagdrop of most weapons by half a second.

### CODE REVIEW

**Comment by Aurx** [comment]
Issue: If the user manually changes mags within the automagdrop_delay_time window, this will drop the new, full mag.  
Solution: Add another check after the spawn statement.

**Comment by Shadowmech88** [comment]
You mean inside the `spawn()`.

**Comment by Aurx** [comment]
I have no clue what the commonly accepted formatting terminology is.  After the call to spawn(), at the beginning of the spawn()'s block.  Between 232 and 233.

**Comment by ultraplen** [comment]
Fixed

File: code/modules/projectiles/guns/projectile.dm [hunk 1]
```diff
@@ -229,10 +229,13 @@
 /obj/item/weapon/gun/projectile/afterattack(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, flag, struggle = 0)
 	..()
 	if(!chambered && stored_magazine && !stored_magazine.ammo_count() && gun_flags &AUTOMAGDROP) //auto_mag_drop decides whether or not the mag is dropped once it empties
+		var/drop_me = stored_magazine // prevents dropping a fresh/different mag.
 		spawn(automagdrop_delay_time)
-			RemoveMag(user)
-			if(mag_drop_sound)
-				playsound(user, mag_drop_sound, 40, 1)
+			if(stored_magazine = drop_me)
```
Comment (Shadowmech88) [inline code/modules/projectiles/guns/projectile.dm]: This won't work correctly.
You need to use `==`

**Comment by ultraplen** [comment]
Fixed and tested, worked as intended.

File: code/modules/projectiles/guns/projectile.dm (hunk 1 above)
Comment (ultraplen) [reply code/modules/projectiles/guns/projectile.dm]: oddly enough it compiled on my machine

Comment (Exxion) [reply code/modules/projectiles/guns/projectile.dm]: Yeah, won't prevent a compile, but will cause unintended behavior

**Pull request merged**